src/time_travel/patchers/base_patcher.py

```diff
@@ -0,0 +1,141 @@
+    def get_events_namespace(cls):
+        """Return the namespace of the patcher's events."""
+        return None
+    
+    @classmethod
+    def get_events_types(cls):
+        """Return Enum of the patcher's events types."""
+        return None
+    
+    def get_patched_module(self):
+        """Do more stuff."""
+        raise NotImplementedError()
+        
+    def get_patch_actions(self):
+        """Return list of patches to do.
+        
+        The list structure is tuples containgin:
+            (real_object_name,
+             the_real_object,
+             fake_object)
+        """
+        raise NotImplementedError()
+        
+    def start(self):
+        """Start the patcher.
+        
+        The logic to the patchers start is based on the work done by:
+        spulec/freezegun
+        under
+        https://github.com/spulec/freezegun
+        
+        Copyright (C) 2017 spulec/freezegun
+        
+        Licensed under the Apache License, Version 2.0 (the "License"); you may
+        not use this file except in compliance with the License. You may obtain
+        a copy of the License at
+        
+          http://www.apache.org/licenses/LICENSE-2.0
+        
+        Unless required by applicable law or agreed to in writing,
+        software distributed under the License is distributed on an
+        "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+        KIND, either express or implied.  See the License for the
+        specific language governing permissions and limitations
+        under the License.
+        
+        Modifications:
+        
+        Modifications to the file was to leave the inner change of the loaded
+        modules and removing any other related logic to a specific module.
+        """
+        patch_actions = self.get_patch_actions()
+        
+        real_id_to_fake = {id(real): fake for _, real, fake in patch_actions}
+        
+        patched_module = self.get_patched_module()
+        
+        # Change modules for later imports.
+        for obj_name, real_obj, fake_obj in patch_actions:
+            self._save_for_undo(patched_module, obj_name, real_obj)
+            setattr(patched_module, obj_name, fake_obj)
+            
+        if self.modules_to_patch:
+            # If only a given list of modules is required to be patched
+            modules = [sys.modules[name] for name in self.modules_to_patch] 
+        else:
+            # not given a specific module to patch on.
+            # Create the list of all modules to search for the patched objects.
+            # Patch on all loaded modules.
+            modules = [
+                module for mod_name, module in sys.modules.items() if
+                inspect.ismodule(module) and
+                hasattr(module, '__name__') and
+                # Don't patch inside this module,
+                # or inside the original module.
+                module.__name__ not in ([patched_module, self.patcher_module,
```

Comment:
move `self.patcher_module` to a new line as well.

Reply:
Don't have to according to pep8, but OK.